### PR TITLE
SAMZA-1408 update master doc to use 0.14.0-SNAPSHOT after 0.13.1 release

### PR DESCRIPTION
**Pull request by fredji97**

### CODE REVIEW

File: docs/archive/index.html [hunk 1]
```diff
@@ -27,6 +27,14 @@ <h4 id="latest">Latest Release</h4>
   <li><a href="../startup/hello-samza/latest">Hello Samza</a></li>
 </ul>
 
+<h4 id="13.1">0.13.1 Release</h4>
+
+<ul class="documentation-list">
+  <li><a href="../learn/documentation/0.13.1">Documentation</a></li>
```
Comment (sborya) [inline docs/archive/index.html]: and who will be creating these docs? Are they created automatically?

File: docs/_config.yml [hunk 2]
```diff
@@ -25,4 +25,4 @@ exclude: [_docs]
 baseurl: http://samza.apache.org
 version: latest
 # this is the version you will go if you click 'switch version' in "latest" pages.
-latest-release: '0.13'
+latest-release: '0.13.1'
```
Comment (nickpan47) [inline docs/_config.yml]: No need to change this. We only track major versions in doc site (i.e. 0.13.1 doc will over write 0.13 version doc, not creating a new 0.13.1 version doc site).

File: docs/_layouts/default.html [hunk 3]
```diff
@@ -103,6 +103,7 @@
             <h1><i class="fa fa-history"></i> Archive</h1>
             <ul>
               <li><a href="/archive/index.html#latest">latest</a></li>
+              <li><a href="/archive/index.html#13.1">0.13.1</a></li>
```
Comment (nickpan47) [inline docs/_layouts/default.html]: Don't need this.

File: docs/archive/index.html [hunk 4]
```diff
@@ -27,6 +27,14 @@ <h4 id="latest">Latest Release</h4>
   <li><a href="../startup/hello-samza/latest">Hello Samza</a></li>
 </ul>
 
+<h4 id="13.1">0.13.1 Release</h4>
```
Comment (nickpan47) [inline docs/archive/index.html]: Same here. Since we are simply over writing 0.13 doc sites, no need to create new versioned doc site links here.

**Comment by fredji97** [comment]
PR updated based on the comments. @nickpan47 please take another look, thanks!

**Pull request closed without merging**